crates/uv/src/commands/project/lock_target.rs

```diff
@@ -268,6 +271,7 @@ impl<'lock> LockTarget<'lock> {
                                 lock.version(),
                             ));
                         }
+                        let lock = lock.with_proxy_urls(proxies)?;
```

Comment:
It seems like this should be composable -- in other words, should we do this outside of `read`, i.e., expect callers to call `with_proxy_urls` rather than passing it in as an argument and calling this method as the last step? It seems more flexible to me. Same for `commit` -- could we expect callers to apply the proxies before calling `commit`?